### ISSUE TITLE
fredoverflow/karel -- Could not locate the karel.jar file from the software provided 

### ISSUE THREAD

**Issue by sharrisn** [fredoverflow/karel#1]
I followed the instructions for installing Karel the Robot but found that there is no "karel.jar" file included. There maybe other files missing too I don't know.  
I hope you can make the corrections because in my opinion, this is the **best** version of Karel the Robot I found to learn Java from.

Thank you for your time and effort in developing this product.

**Comment by StoneColdMagic** [comment]
Has this been resolved yet? I'm having the same issue. I'd love to be able to download a runnable version of karel, but I can't find the .jar either.

Thank you!

Jeff

**Comment by fredoverflow** [comment]
Do you have a [JDK](https://adoptopenjdk.net), [git](https://git-scm.com/downloads) and [Maven](https://maven.apache.org/install.html) installed? What error messages do you get when following the instructions?

    git clone https://github.com/fredoverflow/freditor
    cd freditor
    mvn install
    cd ..
    git clone https://github.com/fredoverflow/karel
    cd karel
    mvn package

**Comment by sharrisn** [comment]
I see at least one other person is having the same problem.  To answer your
question, Yes I installed all required software.  There are no errors but
when I look at the directory of files the .jar file is missing.

I will try reinstalling all the files again and let you know if the file is
still missing.

Thank you.

**Comment by sharrisn** [comment]
Success!
Here is why I originally could not find the karel.jar file:

After using "git clone /https://github.com/fredoverflow/freditor"  to set
up the freditor directory, I issued the command "mvn install".  The
executable "mvn" produced the following error:

Error: Could not find or load main class
org.codehaus.plexus.classworlds.launcher.Launcher

This was perplexing because I knew I installed Maven correctly. Despite the
error message, I issued the command
"git clone https://github.com/fredoverflow/karel"

When I looked under this directory, Karel.jar was not there.  I hadn't
realized that the karel.jar file results from the "mvn package" build.
Obviously, if the "mvn" command is not working properly the build will fail.

I looked into the "mvn" problem.  What I did was locate the source of the
"mvn" executable and found that although it lives in /usr/bin one must use
the complete path to the mvn executable to build everything!
After issuing the complete path with the two (2) commands:  "/usr/bin/mvn
install" and /usr/bin/mvn package" everything worked!

So, in summary, users must be sure to use the complete path to the mvn
executable (at least with Linux), to get everything to work.
Thanks for getting back to me.  I hope this write-up helps anyone else who
may experience problems with installing your version of Karel on Linux.
Sam Harrison

**Comment by fredoverflow** [comment]
Weird. Which Linux distribution are you using, and how did you install Maven?

At least on Debian 10, `sudo apt install maven` installs Maven correctly, and `mvn install` and `mvn package` work correctly without additional paths.

**Comment by sharrisn** [comment]
I am using the latest version of Debian called MX Linux.   It's always up
to date.  Maven installed with *sudo apt install *and there were
"no visible errors" during the installation.  I followed your directions.
As I mentioned, once I issued the full path to mvn (/usr/bin/mvn)  all
remaining files installed correctly.

Thanks again for your help and concern.
Sam Harrison
